unit-tests/src/test/scala/scala/scalanative/regex/ExecTest.scala

```diff
@@ -0,0 +1,655 @@
+package scala.scalanative
+package regex
+
+// import java.io.InputStreamReader
+// import java.io.File
+// import java.io.FileInputStream
+// import java.{util => ju}
+// import java.util.regex.PatternSyntaxException
+// import java.util.zip.GZIPInputStream
+// import java.util.{Arrays, Collections}
+
+// import scala.util.control.NonFatal
+// import scala.util.control.Breaks._
+
+import org.junit.Ignore
+import org.junit.Test
+import org.junit.Assert._
+
+// TestRE2 tests this package's regexp API against test cases
+// considered during (C++) RE2's exhaustive tests, which run all possible
+// regexps over a given set of atoms and operators, up to a given
+// complexity, over all possible strings over a given alphabet,
+// up to a given size.  Rather than try to link with RE2, we read a
+// log file containing the test cases and the expected matches.
+// The log file, re2.txt, is generated by running 'make exhaustive-log'
+// in the open source RE2 distribution.  http://code.google.com/p/re2/
+//
+// The test file format is a sequence of stanzas like:
+//
+//      strings
+//      "abc"
+//      "123x"
+//      regexps
+//      "[a-z]+"
+//      0-3;0-3
+//      -;-
+//      "([0-9])([0-9])([0-9])"
+//      -;-
+//      -;0-3 0-1 1-2 2-3
+//
+// The stanza begins by defining a set of strings, quoted
+// using Go double-quote syntax, one per line.  Then the
+// regexps section gives a sequence of regexps to run on
+// the strings.  In the block that follows a regexp, each line
+// gives the semicolon-separated match results of running
+// the regexp on the corresponding string.
+// Each match result is either a single -, meaning no match, or a
+// space-separated sequence of pairs giving the match and
+// submatch indices.  An unmatched subexpression formats
+// its pair as a single - (not illustrated above).  For now
+// each regexp run produces two match results, one for a
+// ``full match'' that restricts the regexp to matching the entire
+// string or nothing, and one for a ``partial match'' that gives
+// the leftmost first match found in the string.
+//
+// Lines beginning with # are comments.  Lines beginning with
+// a capital letter are test names printed during RE2's test suite
+// and are echoed into t but otherwise ignored.
+//
+// At time of writing, re2.txt is 32 MB but compresses to 760 kB,
+// so we store re2.txt.gz in the repository and decompress it on the fly.
+
+@Ignore("takes too long to run")
+class ExecTest {
+
+  /* The following dummy test was added during the JUnit
+   * port and the rest of the file is commented out because
+   * of compiler errors. Having one test means the @Ignore
+   * on the test works and will show output when run as
+   * part of the rest of the tests.
+   *
+   *
+   * When the rest of the file is uncommented the tests
+   * ran despite of the compile errors but slowly as noted above.
```

Comment:
What does that mean? If there were compile errors, it couldn't possibly run.

Reply:
It says error but prints `[warn]` during the Optimization phase so that makes more sense.
```
sbt:scala-native> tests/testOnly scala.scalanative.regex.ExecTest
[info] compiling 7 Scala sources to /Users/eric/workspace/scala-native/unit-tests/target/scala-2.12/classes ...
[info] compiling 197 Scala sources to /Users/eric/workspace/scala-native/unit-tests/target/scala-2.12/test-classes ...
[info] Linking (2574 ms)
[info] Checking intermediate code (1322 ms)
[info] Dumping intermediate code (linked) (2921 ms)
[info] Discovered 4744 classes and 30631 methods
[info] Optimizing (debug mode) (8577 ms)
[warn] Found 1 errors on @"M19java.io.PrintStreamD6formatL16java.lang.StringLAL16java.lang.Object_L19java.io.PrintStreamEO" :
[warn]   0000  didopt def @"M19java.io.PrintStreamD6formatL16java.lang.StringLAL16java.lang.Object_L19java.io.PrintStreamEO" : (@"T19java.io.PrintStream", @"T16java.lang.String", array[@"T16java.lang.Object"]) => @"T19java.io.PrintStream" {
[warn]   0001  %40000(%1 : @"T19java.io.PrintStream", %2 : @"T16java.lang.String", %3 : array[@"T16java.lang.Object"]):
[warn]   0002    %50002 = ieq[@"T16java.lang.Object"] %1 : @"T19java.io.PrintStream", null
[warn]   0003    if %50002 : bool then %60000 else %70000
[warn]   0004  %70000:
[warn]   0005    jump %180000
[warn]   0006  %60000:
[warn]   0007    jump %160000
[warn]   0008  %160000:
[warn]   0009    %160001 = classalloc @"T30java.lang.NullPointerException"
[warn]   0010    %160002 = fieldstore[bool] %160001 : !?@"T30java.lang.NullPointerException", @"M19java.lang.ThrowableF18writableStackTracePT19java.lang.Throwable", true
[warn]   0011    %160003 = fieldstore[bool] %160001 : !?@"T30java.lang.NullPointerException", @"M19java.lang.ThrowableF17enableSuppressionPT19java.lang.Throwable", true
[warn]   0012    %160004 = call[(@"T19java.lang.Throwable") => @"T19java.lang.Throwable"] @"M19java.lang.ThrowableD16fillInStackTraceL19java.lang.ThrowableEO" : ptr(%160001 : !?@"T30java.lang.NullPointerException")
[warn]   0013    jump %170000
[warn]   0014  %170000:
[warn]   0015    throw %160001 : !?@"T30java.lang.NullPointerException"
[warn]   0016  %180000:
[warn]   0017    %180001 = module @"T17java.util.Locale$"
[warn]   0018    %180002 = call[(@"T17java.util.Locale$") => @"T16java.util.Locale"] @"M17java.util.Locale$D10getDefaultL16java.util.LocaleEO" : ptr(%180001 : !?@"T17java.util.Locale$")
[warn]   0019    if %50002 : bool then %220000 else %230000
[warn]   0020  %220000:
[warn]   0021    %220002 = classalloc @"T23java.lang.StringBuilder"
[warn]   0022    %220003 = arrayalloc[char] int 16
[warn]   0023    %220004 = fieldstore[array[char]] %220002 : !?@"T23java.lang.StringBuilder", @"M31java.lang.AbstractStringBuilderF5valueO", %220003 : !?@"T35scala.scalanative.runtime.CharArray"
[warn]   0024    jump %300000(%220002 : !?@"T23java.lang.StringBuilder")
[warn]   0025  %230000:
[warn]   0026    jump %300000(%1 : @"T19java.io.PrintStream")
[warn]   0027  %300000(%300001 : @"T16java.lang.Object"):
[warn]   0028    %40002 = classalloc @"T19java.util.Formatter"
[warn]   0029    %40003 = fieldstore[@"T20java.lang.Appendable"] %40002 : !?@"T19java.util.Formatter", @"M19java.util.FormatterF4_outO", %300001 : @"T16java.lang.Object"
[warn]   0030    %40004 = fieldstore[@"T16java.util.Locale"] %40002 : !?@"T19java.util.Formatter", @"M19java.util.FormatterF7_localeO", %180002 : @"T16java.util.Locale"
[warn]   0031    %40005 = call[(@"T19java.util.Formatter", @"T16java.lang.String", array[@"T16java.lang.Object"]) => @"T19java.util.Formatter"] @"M19java.util.FormatterD6formatL16java.lang.StringLAL16java.lang.Object_L19java.util.FormatterEO" : ptr(%40002 : !?@"T19java.util.Formatter", %2 : @"T16java.lang.String", %3 : array[@"T16java.lang.Object"])
[warn]   0032    ret %1 : @"T19java.io.PrintStream"
[warn]   0033  }
[warn]   in inst #29 / stored value : 
[warn]     expected @"T20java.lang.Appendable", but got @"T16java.lang.Object"
[warn] 1 errors found
[info] Checking intermediate code (366 ms)
[info] Dumping intermediate code (optimized) (2976 ms)
[info] Dumping intermediate code (lowered) (5226 ms)
[info] Generating intermediate code (10696 ms)
[info] Produced 8 files
[info] Compiling to native code (8967 ms)
[info] Linking native code (immix gc, none lto) (655 ms)
[info] Total (42452 ms)
[info] Starting process '/Users/eric/workspace/scala-native/unit-tests/target/scala-2.12/tests-out' on port '52084'.
skip: NOTE	implicit vs. explicit repetitions : 2009-02-02
  | => tests / Test / nativeLink 42s
skip: NOTE	additional repetition tests graciously provided by Chris Kuklewicz www.haskell.org 2009-02-02

skip: NOTE	null subexpression matches : 2002-06-06

skip: NOTE	all standard compliant implementations should pass these : 2002-05-31

EgrepLiterals.Lowercase

  | => scala.scalanative.regex.ExecTest 29s
```